### ISSUE TITLE
error connecting to DB: pq: password authentication failed for user "listmonk"

### ISSUE THREAD

**Issue by ipflowss** [knadh/listmonk#517]
I am using docker-compose.yml via the install-prod.sh script: sh -c "$(curl -fsSL https://raw.githubusercontent.com/knadh/listmonk/master/install-prod.sh)"

I am getting the following error and not sure how to correct.  I can see the db password in the config.toml and docker-compose.yml file.

```
2021/10/17 23:09:14 main.go:93: v2.0.0 (05585b7 2021-09-27T17:59:39Z)
2021/10/17 23:09:14 init.go:112: reading config: config.toml
2021/10/17 23:09:14 init.go:229: connecting to db: listmonk_db:5432/listmonk
2021/10/17 23:09:14 init.go:232: error connecting to DB: pq: password authentication failed for user "listmonk"
```

```
x-db-defaults: &db-defaults
  image: postgres:13
  ports:
    - "9432:5432"
  networks:
    - listmonk
  environment:
    - POSTGRES_PASSWORD=7nsG2SeOuVjS6
    - POSTGRES_USER=listmonk
    - POSTGRES_DB=listmonk
  restart: unless-stopped
  healthcheck:
    test: ["CMD-SHELL", "pg_isready -U listmonk"]
    interval: 10s
    timeout: 5s
    retries: 6

services:
  db:
    <<: *db-defaults
    container_name: listmonk_db
    volumes:
      - type: volume
        source: listmonk-data
        target: /var/lib/postgresql/data
```

```
# Database.
[db]
host = "listmonk_db"
port = 5432
user = "listmonk"
password = "7nsG2SeOuVjS6"
database = "listmonk"
ssl_mode = "disable"
max_open = 25
max_idle = 25
max_lifetime = "300s"
```

**Comment by xnathanh** [comment]
> I'm testing out the binary installer (no docker) on my linux server and I'm getting this error too even though I put the correct PW in the `config.toml` and can login via `psql -U listmonk -h localhost -p 9432 listmonk`
> 
> ```
> ./listmonk
> 2023/12/01 01:36:27 main.go:102: v2.5.1 (a6a2b69 2023-08-15T15:49:28Z, linux/amd64)
> 2023/12/01 01:36:27 init.go:145: reading config: config.toml
> 2023/12/01 01:36:27 init.go:273: connecting to db: localhost:5432/listmonk
> 2023/12/01 01:36:27 init.go:277: error connecting to DB: pq: password authentication failed for user "listmonk"
> ```
> 
> The default config has `port = 5432`:
> 
> ```
> [app]
> # Interface and port where the app will run its webserver.  The default value
> # of localhost will only listen to connections from the current machine. To
> # listen on all interfaces use '0.0.0.0'. To listen on the default web address
> # port, use port 80 (this will require running with elevated permissions).
> address = "localhost:9000"
> 
> # BasicAuth authentication for the admin dashboard. This will eventually
> # be replaced with a better multi-user, role-based authentication system.
> # IMPORTANT: Leave both values empty to disable authentication on admin
> # only where an external authentication is already setup.
> admin_username = "listmonk"
> admin_password = "listmonk"
> 
> # Database.
> [db]
> host = "localhost"
> port = 5432
> user = "listmonk"
> password = "listmonk"
> 
> # Ensure that this database has been created in Postgres.
> database = "listmonk"
> 
> ssl_mode = "disable"
> max_open = 25
> max_idle = 25
> max_lifetime = "300s"
> 
> # Optional space separated Postgres DSN params. eg: "application_name=listmonk gssencmode=disable"
> params = ""
> ```
> 
> I noticed that I'm able to login via 9432 but not 5432: `psql -U listmonk -h localhost -p 5432 listmonk`, so I changed the `config.toml` port to 9432 and `./listmonk` works now. But is this ok? Should the default be changed? Is it just a minor glitch given that I already tried to install listmonk via docker on this machine and the docker install created it with port 5432? Or does the port in the default binary need to be changed to 9432?

try 
1. docker-compose down -v
2. docker compose up -d db
3. docker compose run --rm app ./listmonk --install
4. docker compose up app

**Comment by MaximilianKohler** [comment]
> try
> 
>     docker-compose down -v
>     docker compose up -d db
>     docker compose run --rm app ./listmonk --install
>     docker compose up app

Thanks, but I'm using the binary, not docker.

I think it was just a leftover fragment from installing listmonk with docker previously. I think the docker install uses both port 9432 and 5432, but the binary install only uses 1 port.

**Comment by satoshinotdead** [comment]
Hello there, I'm facing the same issue using the script and/or managing docker-compose file by my own.

It's the same behavior that I saw a few months ago. I don't like to use the binary but I don't know what's going on.

```
error connecting to DB: pq: password authentication failed for user "listmonk"
```

Tried to remove all volumes, using a simple `listmonk` as password and I'm still with the issue. Any guideline or hint will be appreciated :)
